### PR TITLE
FMDN upstream part 4: Ephemeral Identity Key (EIK) storage

### PR DESCRIPTION
**Pull request by kapi-no**
Added a new (top) commit that adds this storage item to the factory reset test.

### CODE REVIEW

File: tests/subsys/bluetooth/fast_pair/storage/factory_reset/src/main.c [hunk 1]
```diff
@@ -320,6 +368,11 @@ static void fp_storage_validate_empty(void)
 	zassert_ok(err, "Unexpected error in Account Key get");
 	zassert_equal(read_cnt, 0, "Invalid Account Key count");
 
+	ret = fp_storage_eik_is_provisioned();
+	zassert_equal(ret, 0, "Expected unprovisioned EIK");
+	err = fp_storage_eik_get(eik);
```
[the remark this answers is not in the document]
Comment (kapi-no) [reply tests/subsys/bluetooth/fast_pair/storage/factory_reset/src/main.c:373]: This test validates the factory reset functionality. I do not want to sneak in extra functionality here, we should write a new test case similar to Account Keys tests if we want to do that

**Comment by kapi-no** [comment]
Rebased and resolved conflicts that appeared after the Beacon Clock storage item was merged:

https://github.com/nrfconnect/sdk-nrf/pull/15275

File: subsys/bluetooth/services/fast_pair/fp_storage/include/fp_storage_eik.h [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+/** Delete the Ephemeral Identity Key (EIK) from the device storage.
+ *
+ * @return 0 If the operation was successful. Otherwise, a (negative) error code is returned.
+ */
+int fp_storage_eik_delete(void);
+
+/** Check if the Ephemeral Identity Key (EIK) is available in the device storage.
+ *
+ * @return A (negative) error code if the operation fails.
+ *         A positive value if the EIK is available in the device storage.
```
[the remark this answers is not in the document]
Comment (kapi-no) [reply subsys/bluetooth/services/fast_pair/fp_storage/include/fp_storage_eik.h]: improved in the last patch

File: tests/subsys/bluetooth/fast_pair/storage/factory_reset/src/main.c [hunk 3]
```diff
@@ -303,13 +327,37 @@ static void personalized_name_validate_loaded(const char *name)
...
+}
+
+static void eik_validate_loaded(const uint8_t *eik)
+{
+	int err;
+	int ret;
+	uint8_t eik_loaded[FP_STORAGE_EIK_LEN];
+
+	ret = fp_storage_eik_is_provisioned();
+	zassert_true((ret >= 0), "Unexpected error in EIK provisioned state check");
```
[the remark this answers is not in the document]
Comment (kapi-no) [reply tests/subsys/bluetooth/fast_pair/storage/factory_reset/src/main.c]: improved in the last patch

**Review by MarekPieta — APPROVED** [review]

**Review by alstrzebonski — APPROVED** [review]

**Pull request merged**